homeassistant/components/daikin/number.py

```diff
@@ -0,0 +1,186 @@
+"""Support for Daikin AirBase zone temperatures."""
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from homeassistant.components.number import NumberDeviceClass, NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, Platform, UnitOfTemperature
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from . import DOMAIN as DAIKIN_DOMAIN, DaikinApi
+
+ZONE_ICON = "mdi:thermostat"
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Old way of setting up the platform.
+
+    Can only be called when a user accidentally mentions the platform in their
+    config. But even in that case it would have been ignored.
+    """
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Daikin climate based on config_entry."""
+    daikin_api: DaikinApi = hass.data[DAIKIN_DOMAIN][entry.entry_id]
+    temperatures: list[DaikinZoneTemperature] = []
+    if zones := daikin_api.device.zones:
+        temperatures.extend(
+            [
+                DaikinZoneTemperature(daikin_api, zone_id)
+                for zone_id, zone in enumerate(zones)
+                if zone[0] != "-" and zone[2] != 0
+            ]
+        )
+
+    async_add_entities(temperatures)
+
+
+class DaikinZoneTemperature(NumberEntity):
+    """Representation of a zone temperature setting."""
+
+    _attr_icon = ZONE_ICON
```

Comment:
```suggestion
    _attr_icon = "mdi:thermostat"
```

Reply:
Isn't the device_class already proving the right icon in the UI?

Reply:
As this is more of a thermostat I thought this icon is more suitable